### PR TITLE
fix: Display informative error message on deleting event

### LINKED ISSUE #3508: Proper error message should be displayed when deleting an event
**Describe the bug**
When deleting an event associated with orders, it should show the correct error message as it gets from the server.

**Additional context**
On it.

### PR DESCRIPTION
**Pull request by prateekj117**
Fixes #3508 

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [X] My branch is up-to-date with the Upstream `development` branch.
- [X] The acceptance, integration, unit tests and linter pass locally with my changes 
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by prateekj117** [comment]
@uds5501 @mrsaicharan1 @shreyanshdwivedi @kushthedude Please review.

File: app/controllers/admin/events/list.js [hunk 1]
```diff
@@ -132,7 +132,7 @@ export default class extends Controller.extend(EmberTableControllerMixin) {
       this.notify.success(this.l10n.t('Event has been deleted successfully.'));
     } catch (e) {
       console.warn(e);
-      this.notify.error(this.l10n.t('An unexpected error has occurred.'));
+      this.notify.error(this.l10n.t(e.errors[0].detail));
```
Comment (uds5501) [inline app/controllers/admin/events/list.js]: use `tVar` for translation helper

**Comment by prateekj117** [comment]
@uds5501 Done. Please review now.

File: app/controllers/admin/events/list.js [hunk 2]
```diff
@@ -132,7 +132,15 @@ export default class extends Controller.extend(EmberTableControllerMixin) {
       this.notify.success(this.l10n.t('Event has been deleted successfully.'));
     } catch (e) {
       console.warn(e);
-      this.notify.error(this.l10n.t('An unexpected error has occurred.'));
+      if (e.errors[0].detail) {
```
Comment (CosmicCoder96) [inline app/controllers/admin/events/list.js]: what if there is no e.errors defined, the entire thing will crash

**Comment by prateekj117** [comment]
@CosmicCoder96 Done. Please review now.

File: app/controllers/admin/events/list.js [hunk 3]
```diff
@@ -132,7 +132,15 @@ export default class extends Controller.extend(EmberTableControllerMixin) {
       this.notify.success(this.l10n.t('Event has been deleted successfully.'));
     } catch (e) {
       console.warn(e);
-      this.notify.error(this.l10n.t('An unexpected error has occurred.'));
+      if ('errors' in e) {
```
Comment (shreyanshdwivedi) [inline app/controllers/admin/events/list.js]: Is this even the right way to check? I think you should check for length here

Comment (prateekj117) [reply app/controllers/admin/events/list.js]: @shreyanshdwivedi It works as expected. Also, checking the length of `e.errors` will again crash the whole thing as pointed out by @CosmicCoder96

Comment (CosmicCoder96) [reply app/controllers/admin/events/list.js]: what if errors is not an array. errors[0] will crash.

**Pull request closed without merging**